packages/sign-client/src/controllers/engine.ts

```diff
@@ -1028,6 +1029,23 @@ export class Engine extends IEngine {
     return formatMessage(request, iss);
   };
 
+  public processRelayMessageCache: IEngine["processRelayMessageCache"] = () => {
+    // process the relay messages cache in the next tick to allow event listeners to be registered by the implementing app
+    setTimeout(async () => {
```

Comment:
Not sure if it's a good idea to mix setTimeout with an async operation.
Unless I misunderstood, I'd rather isolate the `setTimeout` and do something along the lines of 
```suggestion
   await sleep(50)
   if (this.relayMessageCache.length === 0) return;
   while (this.relayMessageCache.length > 0) {
```

Reply:
why do we need that small timeout?

Reply:
the goal is to process the message after the finish initializing. This allows to setup listeners after the `client.init()`

Reply:
yes, but why 50 and not other number? Is 50 enough in a RN environment?

Reply:
it's arbitrary amount. We can change it to some other value if you have suggestion

Reply:
Shouldn't we use a flag like `client.initialized` then ?